provider/docker_test.go

```diff
@@ -956,9 +999,10 @@ func TestDockerLoadDockerConfig(t *testing.T) {
 			},
 			expectedFrontends: map[string]*types.Frontend{
 				"frontend-Host-test-docker-localhost": {
-					Backend:        "backend-test",
-					PassHostHeader: true,
-					EntryPoints:    []string{},
+					Backend:              "backend-test",
+					PassHostHeader:       true,
+					EntryPoints:          []string{},
+					WhitelistSourceRange: []string{},
```

Comment:
I'd still remove the empty slice from the test case definition and sanitize inside the test body; that is, check if the value is nil and assign the empty slice to it. That keeps the test cases clean and moves the sanitization logic into a central location.

As far as being able to easily spot what's wrong in case of a test failure, you should use something like [go-spew](https://github.com/davecgh/go-spew)'s `Sdump`. We already use it at other occasions, so you should be able to use it in your tests too.